src/LanguageServer/Microsoft.CodeAnalysis.LanguageServer/HostWorkspace/MetadataService.cs

```diff
@@ -24,21 +24,19 @@ public IWorkspaceService CreateService(HostWorkspaceServices workspaceServices)
 
     internal sealed class MetadataService : IMetadataService
     {
-        private readonly IDocumentationProviderService _documentationProviderService;
+        private readonly MetadataReferenceCache _metadataCache;
 
         public MetadataService(IDocumentationProviderService documentationProviderService)
         {
-            _documentationProviderService = documentationProviderService;
+            _metadataCache = new MetadataReferenceCache((path, properties) =>
```

Comment:
Not 100% required for the fix, but enabled reference sharing/caching while I was working in the area.  `FileWatchedPortableExecutableReferenceFactory` can now handle multiple projects using the same instance.